backend/apps/readings/analysis.py

```diff
@@ -53,46 +51,31 @@ def frequency_feelings(self):
         for word in feelings:
             if feelings[word] > 1:
                 frequent_words.append((word, feelings[word]))
-
-        for i in range(len(frequent_words) - 1):
-            for j in range(i + 1, len(frequent_words)):
-                if (frequent_words[i])[1] < (frequent_words[j])[1]:
-                    frequent_words[i], frequent_words[j] = frequent_words[j], frequent_words[i]
-
+        frequent_words.sort(key=lambda x: x[1], reverse=True)
         return frequent_words
 
-    @property
     def context_vs_read_time(self):
         """
-        compares average view times, given different context (ad vs story)
+        Compares average view times, of all contexts
         :return a tuple of the average ad view and the average story view
         """
-        ad_sum = 0
-        ad_count = 0
-        story_sum = 0
-        story_count = 0
 
+        total_contexts_view_times = {}
         for response in self.responses:
             context = response.context.text
```

Comment:
Nearly there, but not quite what I meant -- you can query the db for all contexts by calling `Context.objects.all()` and then iterate through those an in inner loop. Should simplify this a little bit.